InnoGotchi.API/Infrastructure.Services/PetConditionService.cs

```diff
@@ -0,0 +1,218 @@
+﻿using InnoGotchi.Application;
+using InnoGotchi.Application.Contracts.Repositories;
+using InnoGotchi.Application.Contracts.Services;
+using InnoGotchi.Domain.Entities;
+using InnoGotchi.Domain.Enums;
+
+namespace Infrastructure.Services
+{
+    public class PetConditionService : IPetConditionService
+    {
+        private readonly IRepositoryManager _repositoryManager;
+        public PetConditionService(IRepositoryManager repositoryManager)
+        {
+            _repositoryManager = repositoryManager;
+        }
+        public bool IsPetAlive(Pet pet) =>
+            pet.DeathDay == null;
+
+        public async Task UpdatePetsFeedingAndDrinkingLevelsByFarm(Farm farm)
+        {
+            var pets = await _repositoryManager.PetRepository.GetPetsByFarmIdAsync(farm.Id, true);
+            foreach (var pet in farm.Pets)
+            {
+                await UpdatePetFeedingAndDrinkingLevels(pet);
+            }
+            farm.Pets = pets.ToList();
+        }
+
+        public async Task<Pet> UpdatePetFeedingAndDrinkingLevels(Pet pet)
```

Comment:
This methid is too big, it's better to split it in some private methods/ static helper and use them inside this method

Reply:
splitted